### PR TITLE
refactor: extract types to new files

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Extract interfaces and types to `./types.ts` files. This will allow easier v8 coverage migration and improves maintainabillity.
We can also easily exclude types from jest, as they do not exist at runtime anyways (exerpt enums). 

In a later pr i'll refactor enum to type def, so they can be used as type imports.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/cdnjs/types.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+interface CdnjsAsset {
+  version: string;
+  files: string[];
+  sri?: Record<string, string>;
+}
+export interface CdnjsResponse {
```
Comment (viceice) [inline lib/datasource/cdnjs/types.ts:6]:
```suggestion

export interface CdnjsResponse {
```

File: lib/datasource/crate/index.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { privateCacheDir, readFile } from '../../util/fs';
 import { Http } from '../../util/http';
 import * as cargoVersioning from '../../versioning/cargo';
 import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import { CrateRecord, RegistryFlavor, RegistryInfo } from './types';
```
Comment (rarkins) [inline lib/datasource/crate/index.ts:13]:
```suggestion
import type { CrateRecord, RegistryFlavor, RegistryInfo } from './types';
```

?

**Review by rarkins — COMMENTED** [review]
Are all types in `common.ts` files now moved to `types.ts` files? And how about `lib/types/*` - can they be moved closer to where they're used?

File: lib/datasource/crate/index.ts (hunk 2 above)
Comment (viceice) [reply lib/datasource/crate/index.ts:13]: Doesn't work, as `RegistryFlavor` is an enum 😕

**Review by rarkins — APPROVED** [review]

**Pull request merged**